aptos-move/block-executor/src/executor.rs

```diff
@@ -911,21 +908,25 @@ where
 
         let drain_commit_queue = || -> Result<(), PanicError> {
             while let Ok(txn_idx) = scheduler.pop_from_commit_queue() {
-                let end_block = self.materialize_txn_commit(
+                self.materialize_txn_commit(
                     txn_idx,
                     versioned_cache,
                     scheduler_wrapper,
-                    num_txns as u32,
                     start_shared_counter,
                     shared_counter,
                     last_input_output,
                     base_view,
                     global_module_cache,
                     runtime_environment,
+                    total_txns_to_materialize,
                     final_results,
                 )?;
 
-                if !end_block {
+                num_txns_materialized.fetch_add(1, Ordering::SeqCst);
+
+                if num_txns_materialized.load(Ordering::SeqCst)
```

Comment:
both optional, but if you like better:

Nit: we could maybe merge this and do
if num_txns_materialized.fetch_add(1, Ordering::SeqCst) + 1 < total_txns_to_materialize.load(Ordering::SeqCst) {
   continue;
}

supernit: we could also assign to variables and have an else branch where we make sure it's equal (otherwise can return code_invariant_error).

Also pretty sure these can be relaxed (fetch_add anyway does not matter), but it shouldn't matter much and let's just use SeqCst for now.